app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -30,6 +30,20 @@ def fetch_middleware_servers
             @eaps << eap
             server = parse_middleware_server(eap)
 
+            # if immutable or in container flag is not set, check also the agent resource
+            if server.fetch_path(:properties, 'Immutable').nil? || server.fetch_path(:properties, 'In Container').nil?
```

Comment:
You can use `if server.mutable? || !server.fetch_path(:properties, 'In Container')` here, but to make it even better I'd also extract 'In Container' check into a method.